common/scala/src/main/scala/whisk/core/database/ArtifactActivationStore.scala

```diff
@@ -40,9 +40,16 @@ class ArtifactActivationStore(actorSystem: ActorSystem, actorMaterializer: Actor
     implicit transid: TransactionId,
     notifier: Option[CacheChangeNotification]): Future[DocInfo] = {
 
-    logging.debug(this, s"recording activation '${activation.activationId}'")
-
-    val res = WhiskActivation.put(artifactStore, activation)
+    val skipStoring = context.user.limits.disableActivationStore.getOrElse(false)
```

Comment:
This seems awfully late to do and is introducing a hack into the artifact store. IF you want to short circuit the entire activation record, it should be done at the source, not here, in my view... because you've also wasted the effort of gathering logs and creating the record, only to throw it away.

Reply:
It indeed should be done at source. The option says `disableActivationStore`, so we should in theory circumvent the ActivationStore call as a whole.

Reply:
Your comment makes sense to me. I was also thinking where to put it and decided to put it here, to match all occurances in one place.
But I can move it up, of cause.

Reply:
@cbickel it only matches all occurrences for this specific implementation of the ActivationStore though. It shouldn't be in the hand of the activation store to decide if this specific quota is enforced or not.